### PR TITLE
Replace auto focus through persisted focus - group menu

### LINKED ISSUE #12764: Fix cypress tabbing tests and focus for group menu

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts [hunk 1]
```diff
@@ -143,4 +143,17 @@ export class ConfiguratorStorefrontUtilsService {
...
   }
+
+  /**
+   * Persist the keyboard focus state for the given key.
+   * The focus is stored globally or for the given group.
+   *
+   * @param {string | undefined} key - key
+   * @param {string} group? - Group
+   */
+  setFocus(key: string | undefined, group?: string): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]:
```suggestion
 setFocus(key?: string, group?: string): void {
```

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts [hunk 2]
```diff
@@ -330,48 +332,88 @@ export class ConfiguratorGroupMenuComponent {
...
     }
   }
 
+  /**
+   * Persists the keyboard focus state for the given key
+   * from the main group menu by back navigation.
+   *
+   * @param {string | undefined} currentGroupId - Current group ID
+   */
+  setFocusForMainMenu(currentGroupId: string | undefined): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]:
```suggestion
  setFocusForMainMenu(currentGroupId?: string): void {
```

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts [hunk 3]
```diff
@@ -330,48 +332,88 @@ export class ConfiguratorGroupMenuComponent {
...
+  /**
+   * Persists the keyboard focus state for the given key
+   * from the subgroup menu by forwards navigation.
+   *
+   * @param {Configurator.Group} group - Group
+   * @param {string | undefined} currentGroupId - Current group ID
+   */
+  setFocusForSubGroup(
+    group: Configurator.Group,
+    currentGroupId: string | undefined
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Also shorten to 
```suggestion
    currentGroupId?: string 
```
please

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts [hunk 4]
```diff
@@ -330,48 +332,88 @@ export class ConfiguratorGroupMenuComponent {
...
-   * @param {string} currentGroupId - Current group ID
    * @param {Configurator.Group} group - Group
+   * @param {string | undefined} currentGroupId - Current group ID
    * @returns {boolean} - 'true' if the parent group contains a selected group, otherwise 'false'
    */
   containsSelectedGroup(
-    currentGroupId: string,
-    group: Configurator.Group
+    group: Configurator.Group,
+    currentGroupId: string | undefined
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Please shorten

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts]: Done!

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**